plugins/webhooks/index.js

```
@@ -634,53 +646,71 @@ async function createEvents(eventFactory, userFactory, pipelineFactory,
-        } else {
-            /* eslint-disable no-await-in-loop */
-            const token = await p.token;
-            const scmConfig = {
-                scmUri: p.scmUri,
-                token,
-                scmContext
-            };
-            // obtain pipeline's latest commit sha for branch specific job
-            const configPipelineSha = await pipelineFactory.scm.getCommitSha(scmConfig);
-            /* eslint-enable no-await-in-loop */
-            const eventConfig = {
-                pipelineId: p.id,
-                type: 'pipeline',
-                webhooks: true,
-                username,
-                scmContext,
-                startFrom,
-                sha,
-                configPipelineSha,
-                changedFiles,
-                commitBranch: branch,
-                causeMessage: `Merged by ${username}`,
-                meta
-            };
-
-            if (skipMessage) {
-                eventConfig.skipMessage = skipMessage;
-            }
 
-            /* eslint-disable no-await-in-loop */
-            await updateAdmins(userFactory, username, scmContext, p);
-            /* eslint-enable no-await-in-loop */
+            return false;
+        }
+
+        return true;
+    });
 
+    const eventConfigs = await Promise.all(ignoreExtraTriggeredPipelines.map(async (pTuple) => {
+        const pipelineBranch = pTuple.branch;
+        const startFrom = determineStartFrom(action, type, branch, pipelineBranch);
+        const token = await pTuple.pipeline.token;
+        const scmConfig = {
+            scmUri: pTuple.pipeline.scmUri,
+            token,
+            scmContext
+        };
+        // obtain pipeline's latest commit sha for branch specific job
+        let configPipelineSha = '';
+
+        try {
+            configPipelineSha = await pipelineFactory.scm.getCommitSha(scmConfig);
+        } catch (err) {
+            winston.info(`skip create event for branch: ${pipelineBranch}`);
```

what's the implication of ignoring this error ?, do we need to differentiate between branch does not exist vs any error in talking to git ?

---

It intend to ignore 'branch does not exist'. And 5xx error should not be ignored, so I fixed.